### ISSUE TITLE
Reflections on some experiences that might be sorta like community-centered design

### ISSUE THREAD

**Issue by bhaugen** [valueflows/valueflows#169]
Here are some first-draft-quality ramblings about tensions between
dimensions of design, from my experience, that might be related to doing
community-centered design using value flows. This will be a little
long. Sorry. As somebody said, I wasn't smart enough to make it
shorter. But I promised to write something, and thought I better get 
a starting point out there, good or bad.

I say "might be related", because that phrase was not in my vocabulary
when I started out. I knew I wanted to work with real communities, not
make stuff up from my own wonderful ideas (which are often not so
wonderful in practice...) At the same time, I had a focus, which you
can read about here: https://sites.google.com/site/economicgroups/.
That was posted around 2009 when me and Lynn and some other friends
were thinking about economic networks. We had started working on
software for them in 2007.

I also had a lot of experience in manufacturing supply chains, which
are a form of economic network. And in 1996, I decided the
requirements for transitioning to a better economic system required
economic network software. So I spent a year searching for the best
model for such software. Everything I found was limited and inadequate
until I ran into REA (the model I am still using). Bill McCarthy, who
first defined it around 1980, presented it as an internal company
accounting model. But it was obvious to me that the model did not care
about companies at all, and was instead a general-purpose economic
network model. I contacted McCarthy, and told him what I saw, and he
immediately got it and agreed. From there, we started to work
together, until a version of REA was adopted as the ISO Economic and
Accounting Model.

So when I meet a group that might want to work with us, I have some
biases upfront. Are they an economic network? Are they doing something
that might be transformational? And would the REA model more-or-less
fit? (It's flexible, and I'm somewhat flexible about it, but probably
not flexible enough. Or something. More on that below.)

But regardless, to the extent I am doing community-centered design at
all, it's not from an open mind or for all aspects of a community.

You might ask, since I have so many biases, why do I want to learn
community-oriented design? It came from the experience of working with
some really good software developers who do what they call
"user-centered design". And I thought to myself, I don't really do
that or even want to do that. I think first of what this community I
am working with needs (in the economic sphere), not what each
individual user needs. It's not that I am against users. But I have
seen situations where some individual users want stuff that is
damaging to the community as a whole.

And I am just now engaged in an email discussion with the author of
one of the articles on this page which references both community and
user centered design about if and when tensions arise between those
two centers. I think they often do.
http://platform.coop/stories

But I don't think I am a very good designer, nor that I have done a
really good job of community-centered design. Here are some examples
where I think in retrospect I have gone astray, and wonder what I
could have done better or differently. We are also in a similar
situation on a new project

Every time we start to work with a community, we are mostly working
with and through one strong leader who was among the founders and
keeps everything going. These are usually not "natural" but
"intentional" communities and usually mostly virtual, although they
often have a geographic center as well. And usually they see our 
software as an aid to attracting and organizing members to their 
community.

Often we have mostly worked remotely, although we try to meet
community members face-to-face as much as we can. That remoteness
means we are not really members of the community we are working with.
We don't feel their pulse or all their pain.

When I read the stories from @unalee and her associates in design for justice,
it's clear that they tend to work with more "natural" communities, and also 
to immerse themselves more, and have more participation, than I have 
been able to make happen. Or even know how to make happen. 
Software development has some roadblocks to broad participation.
Which is one of the reasons we like Value Flows: might make it easier.

The combination of interaction through a single leader or small group
of leaders, and working remotely, causes problems. One community we
worked with had a leader who described everything in terms of the
future he wanted, not the reality that existed now. It wasn't until we
visited ourselves that we understood what was really happening.

Another problem, which I want to focus on here, is connected to coming
in with a model upfront: this has its good and bad sides, which
affects Value Flows now, and will affect Value Flows implementations
in different communities going forward.

In one community, the leader wanted to design his own ontology. I
wouldn't work from his ontology, partly because I didn't think it
would work, and partly because I already had an ontology and a lot of
code based on it and he wanted something up and running fast. So I
would either have backed out of the project, or he would need to go
along with my ontology.  I don't think I handled it well, though,
because it was always a sticking point in the back of his mind and
came up implicitly and explicitly in arguments.

When we met more community members, they didn't care about that, they
were happy to see some tools being developed to solve some of their
problems. And even the leader was happy because the software did
embody most of his ideas, just not his ontology.

Many members of the community were unhappy with another aspect of
design, though: the user interface of our software does not meet the
Applelish fit and finish standards and user-friendliness they have
become accustomed to from their smart phones. So that's one of the
tensions between community-centered and user-centered design: both
take a lot of time, and different sorts of work. We were working with
another group at the same time that focused on user-centered design,
and it took them a month to do what we could do in a day, but the
results were beautiful. We needed features and functions to cover a
broad range of needs of an economic network, while they could focus on
a single feature, the visualization of a network.

Another version of the ontology clash is happening in a new project.
This time, it is a community member who has developed an ontology he
wants to use. Also this time, his ontology is a lot better-designed
than the first time. It may be mergeable with our ontology. In other
words, the system could work, and he could use it from the viewpoint
he wants.

Ontology as I am using it here means the set of concepts and
relationships in the underlying model and logic of a system. For an
economic system, it means, how the economic relationships work in the
software. It is a critical aspect, and what we focus on when we
develop economic network software.

In this new project, we are also lucky that the person who has his own
ontology also can code, and can take over a lot of the user interface
work. So we are trying to let him run with his ideas and see what happens.
We're hoping it leads to more participation, rather than less.

But the tension I see there is between experience and expertise and
community participation. I am describing some experiences in my
pre-valueflows software projects, but they do now and will in future
affect both the development of the value flows vocabulary and
protocols, and also the use of them in community projects.

In my previous projects, we were working with monolithic software
which made it really difficult for other people to fully participate
in the design and construction of the system.

They needed to either trust us or learn to code in a complex suite of
software. We are now working with a couple of new projects where
people are doing that, and are also therefore able to participate a
lot more fully in the design of the software for their community.
Using their own programmers, with some help from us, they can start
taking over the development of their own system.

But there is still a tension between expertise in the economic model
and logic, expertise in coding in general and in this particular
software suite, and the larger number of people who cannot participate
in those ways.

And expertise in the logic of an economic network. It's not freeform.
It needs to work, and work over time, and also be adaptable.

As I have explained before, we work in spirals between the general and
the particular, spiraling into a particular network with all its
quirks, and then spiraling back out to summarize what we learned from
that experiment to prepare for the next one.

Value Flows has been our form of summarization for the past few years.
One of the lessons we are trying to learn from those monolithic
software experiments is to break up the model into smaller components
which can be mixed and matched as required for a particular community,
and also to be developed independently by different people.

So I envision going forward with Value Flows as a set of components,
the mythical software lego blocks. And wondering if those lego blocks
will enable actually participatory community-centered design. And 
also have enough built-on logic so the lego block constructions 
don't fall apart.

As for Value Flows as its own little community, I am personally delighted 
that Lynn and Pavlik are taking the lead and I can step back, although 
I do sometimes feel the urge to argue about things. 

What do you think? Was this too confused? Did you understand the main
ideas? What do you think were the ideas that are useful?

**Comment by gcassel** [comment]
Thanks very much for sharing these thoughtful reflections.   They make a lot of sense to me, and resonate with my experiences with various communities.

I guess that my main thought about community-centered design is that inclusive tools and techniques should be made (and maintained to be) as simple as reasonably possible.  Difficult learning curves, and mandatory investments of personal training (or other resources) *insulate* community insiders from potential participants.  I think it's the wrong kind of insulation.

I applaud your focus on the idea of components (or modules) which should be usable in part or whole.  I'm quite confident in the ability of Lynn, Pavlik and you @bhaugen to integrate truly reasonable and valuable ideas from others who show up.

**Comment by bhaugen** [comment]
> I guess that my main thought about community-centered design is that inclusive tools and techniques should be made (and maintained to be) as simple as reasonably possible. Difficult learning curves, and mandatory investments of personal training (or other resources) insulate community insiders from potential participants. I think it's the wrong kind of insulation.

I agree with all that, but it is sometimes very difficult to do in practice, especially when the community wants some fairly sophisticated features and then adds on a bunch more complications as additional feature requests. And as I wrote, I am not the world's greatest designer in the first place. It's one of the things I don't think I did very well sometimes, and hope to do better as we all learn.

**Comment by gcassel** [comment]
> ...especially when the community wants some fairly sophisticated features and then adds on a bunch more complications as additional feature requests. 

Yeah, I think that often relates to our advertising/promotion/persuasion based consumer & commercial culture...  charismatic people sure know how to talk new ideas up.   

However, features aren't a bad thing per se, especially if they're optional.

The Stones come to mind...   "You can't always get what you want, but if you try sometimes well you might find 
You get what you need"

**Comment by bhaugen** [comment]
I think *most* of the complications that I am thinking of were justified. It's like, ask somebody to help with this work, or I already did it but it wasn't planned, or I want to help with this but nobody asked me, etc etc.

This is the context of some work that is usually planned and people commit to do it. But then all kinds of exceptions happen. And in programming, and also apparently in UI, it's the exceptions that kill you.

**Comment by bhaugen** [comment]
By the way, luddite that I am, I finally got my first smart phone. It is a nightmare of frustration. It is not at all intuitive *to me*. (Which is the usual problem with "intuitive design", according to whose intuition?) I missed the first, second and probably 3rd and 4th generation of smart phone UIs, so what is intuitive to the kids is a complete mystery to me...

**Comment by gcassel** [comment]
I believe that community-centered design is, or ought to be, about creating *tools and techniques* which are useful to specific communities-- and, I believe it's deeply important to distinguish *between* tools and techniques.

Often, a specific person or perhaps even a whole community may desire to possess a tool which is perfectly customized to perform a specific, complex action by using an extremely *simple* technique.  This may be because they have to perform that action frequently, or because it's crucially important for them to perform that action at least a couple of times.  It may also be because one or more persons is really fussy and demanding, and doesn't think much about the costs of research, development and support of technologies (including software applications.)   

I suggest that it's often more appropriate to create reasonably generic and flexibly modular tools which can be used separately or together with a wide variety of reasonably simple (but often slightly complex, and *sequential*) techniques.

Ultimately, I think it's important to create *human-centered* design which enables communities to customize flexible tools and recommended techniques (or recipes) according to each community's needs.   I think it's also important for communities to enable their participants to customize the community tools and techniques according to personal preferences, within acceptable community-defined specifications/ parameters for each tool or technique in question.

I think actually a lot of what's called "user centered design" is really meant to be generically human-centered design, but the discussions are often muddled by vague and unclear terminology.   Also, a lot of "user centered design" is done to create proprietary ownership of one or more tools and techniques.   That creates privilege in the subsequent research and development, and that fosters corruption.  We see this often, of course:   for instance, in "free" social media platforms which gradually monetize the participation of their users.

These are fairly loose and unstructured reflections.  I hope they seem appropriate for this reflective topic.

**Comment by coopchange** [comment]
'Human-centered design' seems close; I think 'Community-centered design' is
proposed in part to communicate a need for including Restorative Justice
principles of "all those affected" or "all stakeholders", and also to be a
vehicle for practices from Participatory Action Research or Co-operative/
Collaborative Inquiry (http://tiny.cc/coInquiry) where clear paths and
invitations to participation are established for all stakeholders/ those
affected.

**Comment by unalee** [comment]
Thanks for this great discussion.

I’m not sure if I shared this before, but here is a talk I gave that attempted to distinguish between human centered and community centered design. 

http://civictech.ca/2016/01/13/hacknight-24-jan-12/ (video + slides)

**Comment by bhaugen** [comment]
@unalee thank you very much. Especially for the slides. I found it difficult to understand what you were saying on the video (but Lynn did better - my hearing is pretty bad, and hearing aids don't always help enough). So the slides were really helpful in following the process. And I loved all the details about your toolset. Will say more about them later after I dig in more.

We've never done a deep immersion like that. Never been able to, couldn't afford it. We are thinking of a project in our geographic region where we could do more of that. We did a week in Madison recently for the Mutual Aid Network, but it's not enough. We'd need to live there for some months, I think.

I'm thinking about how people could follow a process something like that coming out of value flows. 

I'd also like to see a more participatory approach to system development. The technical skill requirements tend to generate an elite that produces the system and a lot of other people who can mostly just be consumers. Can we put together a toolset, a set of components that can be plugged together easily enough to cut down on the skill requirements?

And can we develop a process like what Una demonstrated that communities can use to work with the toolset?

**Comment by bhaugen** [comment]
The difference between community-centered design and user-centered design was typified for me in an incident where a particular individual who shall not be named wanted a feature designed just to make him happy. The community requirement here was for this person to enter some data for the community's benefit, that did not benefit the individual in any other way than improving the community. So Lynn put a lot of work into designing a feature that was easy enuf for this individual's tastes, and then he never used it, because he did not give a shit about the community requirement, he just wanted to blow smoke.

Human-centered design is probably something else again. Probably not my bad example...

**Comment by bhaugen** [comment]
@gcassel 
> Often, a specific person or perhaps even a whole community may desire to possess a tool which is perfectly customized to perform a specific, complex action by using an extremely simple technique. This may be because they have to perform that action frequently, or because it's crucially important for them to perform that action at least a couple of times.[...]

> I suggest that it's often more appropriate to create reasonably generic and flexibly modular tools which can be used separately or together with a wide variety of reasonably simple (but often slightly complex, and sequential) techniques.

We've seen that pattern cut both ways, and lean toward thinking the community is probably right. Our software has learned a lot from use cases that did not quite fit what we had before. This example will only make complete sense to people who know our NRP software, but the workflow recipes came from a Guerrilla Translators use case. They required an agonizing several weeks of redesign and retesting. But then, several other networks decided they loved them, for some situations, too. 

That's partly why we do so many use cases in value flows.

**Comment by gcassel** [comment]
>We've seen that pattern cut both ways, and lean toward thinking the community is probably right.

I suggested that it's often more appropriate to create generic tools, but I didn't mean to imply that communities don't deserve perfectly customized tools.  We both care about what specific communities want.  

I further suggest that it's hard to tell what a community wants.  I don't think that the appointed or self-appointed leaders of a community usually have a great grasp of it.   They know a lot, but they have biases due to the ways in which their power (coercive authority and/or social capital) has worked for them-- and, if they control the pursestrings, they distort R&D processes.

I think that human-centered design should (try to) support the intentional design of communities *and*, to whatever extent said communities allow it, expressive power for each type of participant within each community.   Could this perception of the relationship between *communities* and their *participants* help to transcend the conflict between "user-centered" and "community-centered" design?

Of course, it's tremendously hard for communities to fairly design for "types of participants" who have defined rights and responsibilities.    I think that's a deep social challenge, and I want to do my best to help get technology out of the way of it.

**Comment by bhaugen** [comment]
> I suggested that it's often more appropriate to create generic tools, but I didn't mean to imply that communities don't deserve perfectly customized tools. We both care about what specific communities want.

I figured we basically agreed. Part of the point here is that generic tools can become too rigid. For example, REA, the economic framework we use, is just about as simple and flexible as you can get, but we find that some of its assumptions are not compatible with VF, which [I wrote about here](https://docs.google.com/document/d/1YpFaytWfeuGMxZWW6yY59ZTfL0PseAQ_LxqjjBhGogs/edit?usp=sharing), and so we change them. In the process, [the REA community is becoming influenced by VF](https://drive.google.com/file/d/0BxlqOeaPnXHfTkpPRlFNbzcxR0k/view?usp=sharing).

I got acquainted with this problem when I was working on manufacturing systems in the 1980s. The systems at the time had been standardized by APICS, the American Production and Inventory Control Society. But the standardized model did not work for some situations (food processing, for example). So a whole new set of systems was developed for those industries. The VF model supports both, which a bunch of people figured how to do in the mid-1990's.

> I further suggest that it's hard to tell what a community wants. I don't think that the appointed or self-appointed leaders of a community usually have a great grasp of it. 

Yup. We see that all the time. It is one of the flaws of working remotely and not being immersed. I hope if we can provide tools that allow communities to design/configure their own systems, that might help, but we'll see...
